types/index.d.ts

```diff
@@ -0,0 +1,219 @@
+     * Used in the stringify phase in the process and invoked with a `Node` and `VFile` representation of the document to stringify.
+     *
+     * `Compiler` can be a normal function in which case it must return a `string`: the text representation of the given syntax tree.
+     *
+     * `Compiler` can also be a constructor function (a function with keys in its `prototype`) in which case it’s invoked with `new`.
+     * Instances must have a `compile` method which is invoked without arguments and must return a `string`.
+     */
+    Compiler: CompilerFunction | typeof Compiler
+
+    /**
+     * Transform a syntax tree by applying plugins to it.
+     *
+     * @param node
+     * @param file `VFile` or anything which can be given to `vfile()`
+     * @param done Invoked when transformation is complete.
+     * Either invoked with an error or a syntax tree and a file.
+     * @returns `Promise` if `done` is not given. Rejected with an error, or resolved with the resulting syntax tree.
+     */
+    run(node: Unist.Node): Promise<Unist.Node>
+    run(node: Unist.Node, file: vfile.VFile): Promise<Unist.Node>
+    run(node: Unist.Node, done: RunCallback): void
+    run(node: Unist.Node, file: vfile.VFile, done: RunCallback): void
+
+    /**
+     * Transform a syntax tree by applying plugins to it.
+     *
+     * If asynchronous plugins are configured an error is thrown.
+     *
+     * @param node
+     * @param file `VFile` or anything which can be given to `vfile()`
+     * @returns The given syntax tree.
+     */
+    runSync(node: Unist.Node, file?: vfile.VFile): Unist.Node
+
+    /**
+     * Process the given representation of a file as configured on the processor. The process invokes `parse`, `run`, and `stringify` internally.
+     * @param file
+     * @param done Invoked when the process is complete. Invoked with a fatal error, if any, and the VFile.
+     * @returns `Promise` if `done` is not given.
+     * Rejected with an error or resolved with the resulting file.
+     */
+    process(file: vfile.VFile | string | Buffer): Promise<vfile.VFile>
+    process(file: vfile.VFile | string | Buffer, done: ProcessCallback): void
+
+    /**
+     * Process the given representation of a file as configured on the processor. The process invokes `parse`, `run`, and `stringify` internally.
+     *
+     * If asynchronous plugins are configured an error is thrown.
+     *
+     * @param file
+     * @returns Virtual file with modified contents.
+     */
+    processSync(file: vfile.VFile | string | Buffer): vfile.VFile
+
+    /**
+     * Get or set information in an in-memory key-value store accessible to all phases of the process.
+     * An example is a list of HTML elements which are self-closing, which is needed when parsing, transforming, and compiling HTML.
+     *
+     * @param key Identifier
+     * @returns If getting, the value at key
+     */
+    data(key: string): unknown
```

Comment:
[`data(): Object<string: unknown>`](https://github.com/unifiedjs/unified/blob/7dc311b1aa07d950549965c883e6fc1b1ed884a2/index.js#L133-L157) or something is also supported. But apparently not documented, so makes sense to ignore it for now. Or we need to document it as well.

Reply:
I'll update document too. 👍